src/index.ts

```diff
@@ -82,6 +85,126 @@ export const AWS: SDK = new Proxy({} as any, {
           };
         }
 
+        function createSnsHandler(methodName: string) {
+          return async (input: any) => {
+            const url = new URL(endpoint);
+            for (const [k, v] of Object.entries(input)) {
+              if (k !== "PublishBatchRequestEntries") {
+                url.searchParams.set(k, v as any);
+              } else {
+                for (const [i, entry] of (v as any[]).entries()) {
+                  for (const [k, v] of Object.entries(entry)) {
+                    url.searchParams.set(
+                      `PublishBatchRequestEntries.${k}.${i}`,
+                      // `PublishBatchRequestEntries.${k}.${i + 1}`,
+                      // `PublishBatchRequestEntry.${i + 1}.${k}`,
+                      // `PublishBatchRequestEntries.${i + 1}.${k}`,
+                      v as any,
+                    );
+                  }
+                }
+              }
+            }
+            url.searchParams.set("Action", resolveAction(methodName));
+            url.searchParams.set("Version", "2010-03-31");
+
+            const response = await sendRequest(url, {
+              method: "POST",
+              body: JSON.stringify(input),
```

Comment:
does SNS accept a body and url search params?

Reply:
This is likely a remnant of copy-pasting.